pallets/vector/src/storage_utils.rs

```diff
@@ -81,19 +84,42 @@ pub fn get_storage_root(
 			.data()
 			.map_err(|_| StorageError::StorageValueError)?;
 
-		let storage_root = H256::from_slice(item);
+		let storage_root = padded_value(item).map_err(|_| StorageError::StorageValueError)?;
 
 		Ok(storage_root)
 	} else {
 		Err(StorageError::StorageValueError)
 	}
 }
 
+/// padded_value it converts rlp value to H256 which requires 32 bytes and
+/// appends leading zeros if value is less than 32 bytes, otherwise throws an error.
+fn padded_value(rlp_value: &[u8]) -> Result<H256, StorageError> {
+	let mut slot_value = [0u8; 32].to_vec();
+
+	if rlp_value.len() < 32 {
```

Comment:
may consider using a match here

Reply:
clippy has a suggestion for this but seems more complex than `if` `else`

Reply:
Should be very similar current code, anyway, it's your call